packages/synapse-react-client/src/components/EntityTreeTable/components/IdColumnHeader.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { HeaderContext } from '@tanstack/react-table'
+import { EntityBundleRow } from '../EntityTreeTable'
+import ColumnHeader from '../../TanStackTable/ColumnHeader'
+import { CopyToClipboardIcon } from '../../CopyToClipboardIcon'
+
+export const IdColumnHeader: React.FC<
+  HeaderContext<EntityBundleRow, unknown>
+> = props => {
+  const { table } = props
+
+  // Get all visible rows and extract their entity IDs
+  const getVisibleIds = () => {
+    const visibleRows = table.getRowModel().rows
+    return visibleRows
+      .filter(row => !row.original.isLoadMore && row.original.entityId) // Exclude "Load More" rows and rows without entityId
+      .map(row => row.original.entityId)
+      .join('\n')
+  }
+
+  const copyButton = <CopyToClipboardIcon value={getVisibleIds()} sizePx={16} />
```

Comment:
Let's memoize if you're going to compute it each render (unless there's a way to defer computing it until `onClick` of the CopyToClipboardIcon)
```suggestion

  // Get all visible rows and extract their entity IDs
  const visibleIds = useMemo(() => {
    const visibleRows = table.getRowModel().rows
    return visibleRows
      .filter(row => !row.original.isLoadMore && row.original.entityId) // Exclude "Load More" rows and rows without entityId
      .map(row => row.original.entityId)
      .join('\n')
  }, [table.getRowModel().rows])

  const copyButton = <CopyToClipboardIcon value={visibleIds} sizePx={16} />
```